### ISSUE TITLE
Bug Report: Example not working in `allometry::AllomAve`

### ISSUE THREAD

**Issue by SimonGoring** [PecanProject/pecan#1130]
## Description

Running the example in the help file for `AllomAve`:

```R
if(TRUE){                                # Note, I changed the flag so it would run(!)
  pfts = list(FAGR = data.frame(spcd=531,acronym='FAGR'))
  allom.stats = AllomAve(pfts,ngibbs=500)

  ## example of a PFT with multiple species (late hardwood)
  ## note that if you're just using Jenkins the acronym column is optional
  pfts = list(LH = data.frame(spcd = c(531,318),acronym=c('FAGR','ACSA3')))
}
```

Returns the following:

```
[1] "writing output to"                                
[2] "C:/Users/Simon Goring/Documents/GitHub/PalEON-FIA"
$spcd
[1] 531

$acronym
[1] FAGR
Levels: FAGR

[[3]]
[1] 6

[1] "Dropping allom rows: "
Error in (s1 <- s2) <- 0.1 : object 's1' not found
In addition: Warning message:
In nu(allom$parm$Component.ID) : NAs introduced by coercion
```

I propose that this be fixed.

## Context
My `sessionInfo()`:

```R
R version 3.3.1 (2016-06-21)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] tools     stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] PEcAn.allometry_1.4.8 mvtnorm_1.0-5         MCMCpack_1.3-7       
[4] MASS_7.3-45           coda_0.18-1           dplyr_0.5.0          
[7] knitr_1.14           

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.7        lattice_0.20-34    digest_0.6.10      withr_1.0.2       
 [5] assertthat_0.1     grid_3.3.1         R6_2.2.0           MatrixModels_0.4-1
 [9] DBI_0.5-1          magrittr_1.5       lazyeval_0.2.0     SparseM_1.72      
[13] Matrix_1.2-7.1     devtools_1.12.0    mcmc_0.9-4         memoise_1.0.0     
[17] quantreg_5.29      tibble_1.2        
```

**Comment by SimonGoring** [comment]
FYI - It's commit ffd75f5dd4653e6b27f799e8f742e201da49df57 where @bpbond changed `s1 <- s2 = 0.1` to `(s1 <- s2) <- 0.1`.  The brackets break the order of operations, and since `s2` is not yet defined everything screws up.

**Comment by bpbond** [comment]
Huh, interesting screwup by `formatR`. Thank you @SimonGoring ! I'll go look to make sure there aren't any other such 'fixes' anywhere.

**Comment by tonygardella** [comment]
#1131 has fixed this. Thank you @SimonGoring and @bpbond!

**Comment by SimonGoring** [comment]
FWIW - I find [`lintr`](https://cran.r-project.org/web/packages/lintr/index.html)'s suggestions are just as good but a little more hands-off.

**Comment by bpbond** [comment]
Thanks @SimonGoring - I have looked at that too. Really cool package.

**Comment by dlebauer** [comment]
Wow the lintr README is impressive https://github.com/jimhester/lintr/blob/master/README.md

Highly configurable and one example shows how to add a test using `expect_lint_free()`

**Comment by bpbond** [comment]
👍  I think it's been integrated into dev versions of RStudio, so will show in real time.

**Comment by yihui** [comment]
@SimonGoring Just to clarify, `s1 <- s2 = 0.1` exactly means `(s1 <- s2) <- 0.1`, because `<-` has higher precedence than `=`, as I explained in https://github.com/yihui/formatR/issues/59. The order of operations was not actually broken.

**Comment by SimonGoring** [comment]
Right, I looked into the history a bit more and found the offending change, which was in 9a11080bafb2e4812e25430edb7509ef402f4403, where the operation

`s1 = s2 = 0.1` was changed to `s1 <- s2 = 0.1`, which is what actually broke the function.  Then `formatR` cleaned up the code by adding the brackets (which also failed), and converted the `=` to a `<-`.  So, not a `formatR` problem.  A @bpbond problem :)
